web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-events.php

```diff
@@ -0,0 +1,173 @@
+            if ($to && $event['time'] > $to) {
+                return false;
+            }
+            // The event is within the time range.
+            return true;
+        });
+
+        return array_values($filtered_events);
+    }
+
+    /**
+     * Get track events.
+     * 
+     * Accepts optional arguments for post_id, from, and to.
+     * 
+     * @param int|null $post_id The post ID.
+     * @param int|null $from The start time.
+     * @param int|null $to The end time.
+     * 
+     * @return array
+     */
+
+    public function getTrackEvents(int | null $post_id = null, int | null $from = null, int | null $to = null): array
+    {
+
+        /**
+         * A post_id was passed, so we only need to get the details for that post.
+         */
+        if ($post_id) {
+            $all_details = get_metadata('post', $post_id, $this->event_details_field);
+
+            return [$post_id => $this->filterTrackEvents($all_details, $from, $to)];
+        }
+
+        /**
+         * A post_id was not passed, so return results for multiple posts.
+         */
+
+        // The base query arguments.
+        $wp_query_args = [
+            'fields' => 'ids',
+            'posts_per_page'    => -1,
+            'post_type' => ['post', 'page', 'news', 'note-from-antonia'],
+            'post_status' => ['publish', 'pending'],
+            'meta_query' => [
+                'relation' => 'AND',
+                [
+                    'key' => $this->event_timestamp_field,
+                    'compare' => 'EXISTS',
+                ],
+                [
+                    'key' => $this->event_timestamp_field,
+                    'compare' => '!=',
+                    'value' => ''
+                ]
+            ]
+        ];
+
+        // Add clause for date from.
+        if ($from !== null) {
+            $wp_query_args['meta_query'][] = array(
+                'key'     => $this->event_timestamp_field,
+                'value'   => $from,
+                'compare' => '>=',
+                'type'    => 'NUMERIC'
+            );
+        }
+
+        // And, clause for date to.
+        if ($to !== null) {
+            $wp_query_args['meta_query'][] = array(
+                'key'     => $this->event_timestamp_field,
+                'value'   => $to,
+                'compare' => '<=',
+                'type'    => 'NUMERIC'
+            );
+        }
+
+        // Get the posts.
+        $posts = new WP_Query($wp_query_args);
+
+        // Create an array to store the filtered track events.
+        $all_post_events = [];
+
+        foreach ($posts->posts as $post_id) {
+            $all_events = get_metadata('post', $post_id, $this->event_details_field);
+            $all_post_events[$post_id] = $this->filterTrackEvents($all_events, $from, $to);
+        }
+
+        return $all_post_events;
+    }
+
+    // TODO: lifecycle policy, delete events older than x? Or keep only the most recent x events per post?
```

Comment:
@wilson1000 fyi, I've added setters & getters for the tracking. These are the todos - I'll pick them up next.